### ISSUE TITLE
MSE vs CrossEntropy as the loss function. 

### ISSUE THREAD

**Issue by yongduek** [msminhas93/DeepLabv3FineTuning#1]
Thanks for the code sharing, first of all. 

The segmentation output is the image of class ids. So BCELoss (for binary segmentation) or CrossEntropy should be used, I guess, even though MSE also may be used too. 

E.g., this uses crossentropy: https://github.com/wkentaro/pytorch-fcn/blob/master/torchfcn/trainer.py

Any special reason for MSE? I am just curious. 

Thanks.

**Comment by msminhas93** [comment]
You are absolutely right DeepLabv3 uses cross entropy for training. The choice of MSE was related to my current research work (which involved only binary segmentation outputs). For multi-class segmentation, CrossEntropy would make more sense.
